ovirtlib4/system_service.py

```diff
@@ -64,8 +70,121 @@ def _get_collection_entity(self):
 
-    def get(self, *args, **kwargs):
-        """ Same as list() """
+    def _get_wait_for_type(self, wait_for):
+        """
+        Detect the given 'wait_for' type
+
+        Args:
+            wait_for (bool/callable/str/None): object from type of string, function name or boolean
+
+        Returns:
+            bool/callable/str: If the type of given 'wait_for' if detected, None otherwise
+        """
+        try:
+            if type(wait_for) == bool:
+                return bool
+        except Exception:
+            pass
+
+        try:
+            if callable(wait_for):
+                return callable
+        except Exception:
+            pass
+
+        try:
+            if type(wait_for) == str:
+                return str
+        except Exception:
+            pass
+
+        return None
+
+    def _process_sample(self, wait_for, wait_for_method, sample):
+        """
+        Handle a result sample according to the given 'wait_for' and 'wait_for_method'
+
+        Args:
+            wait_for (object): see run_sampler() docstring
+            wait_for_method (str): see run_sampler() docstring
+            sample (list): List of SDK entities, returned by get()
+
+        Returns:
+            list: the given sample if it match, otherwise an empty list
+        """
+        wait_for_type = self._get_wait_for_type(wait_for=wait_for)
+
+        if wait_for_type == bool:
+            if (sample is not []) == wait_for:
+                return sample
+
+        if wait_for_type == callable:
+            results = eval("wait_for(sample)")
+            if results:
+                return sample
+
+        if wait_for_type == str:
+            results = []
+            for obj in sample:
+                results.append(eval(f"obj.{wait_for}"))
+
+            if eval(f"{wait_for_method}(results)"):
+                return sample
+
+        return []
+
+    def run_sampler(self, wait_for, wait_method="all", timeout=5, sleep=1, *args, **kwargs):
+        """
+        Sample the list() method
+
+        Args:
+            wait_for (object): The algorithm to detect a successful method
+                If 'bool': examine the list() returned value,
+                    True - wait until list() will return a non-empty list
+                    False - wait until list() will return an empty list
+                If 'str': 'wait_for' will be examine for each field at the list,
+                    wait until any or all fields are True (depend on 'wait_for_method')
+                If 'function': Wait until given function return True
+            wait_for_method (str): Can be "any" or "all" available only if given 'wait_for' value is string
+                "all": all object at the list should match before exit the wait state
+                "any": exit the wait state for first match
+            timeout (int): Timeout to wait for success in seconds
+            sleep (sleep): Sleep interval between the samplers in seconds
```

Comment:
I think it will be better to change it to "interval" instead of "sleep" which is a known method

Reply:
Mmm, actually yes, but I try to keep the original code as it was in ART, 
'sleep' is much simple, short, easy to remember and the user immediately know what it does 
while 'interval' is longer and needs to be explained, actually it should be called 'sleep_interval' but then it
will very long

Reply:
in your code, you can't do the basic import like:
from time import sleep
because your code will overwrite the sleep method. 
so I think it's better to fix the wrong code here and not continue with the mistaken code.

Reply:
usually, you write:
import time
time.sleep(5)
WDYT?

Reply:
'timeout' was renamed to 'wait_timeout'
and 'sleep' was renamed to 'wait_interval'
To eliminate a conflict with the SDK list() parameters
and to match the 'wait_for' main parameter